transcoder.go

```diff
@@ -118,7 +115,15 @@ func (t *Transcoder) registerService(svc *Service, svcOpts serviceOptions) error
 	}
 
 	if svcOpts.resolver == nil {
-		return fmt.Errorf("service %s is configured with a nil type resolver", svc.schema.FullName())
+		if canUseGlobalTypes(svc.schema) {
+			svcOpts.resolver = protoregistry.GlobalTypes
+		} else {
+			res, err := resolverForService(svc.schema)
+			if err != nil {
+				return fmt.Errorf("failed to create default resolver: %w", err)
+			}
+			svcOpts.resolver = fallbackResolver{res, protoregistry.GlobalTypes}
```

Comment:
Why a fallback resolver and not just return the resolver that resolves? Maybe this could be moved to register method and can then return a better error?
```
unable to registry method: /my.Service/Method/: unable to resolve type: `my.service.MethodRequest`
```

Reply:
The custom resolver only knows about the transitive dependencies of the file that defines the file (or, at the least the request and response types). That's not necessarily that useful for things like `Any` messages, particularly error details, or extensions. So this resolver is effectively the dynamically resolved things for the service _plus_ anything the program already knows about, such as well-known types and the like.

Reply:
Also, that example error message doesn't appear to be related to this error -- this isn't about resolving the message type, but about creating the default resolver.

> I think you can use [dynamicpb.NewMessageType](https://pkg.go.dev/google.golang.org/protobuf/types/dynamicpb#NewMessageType) if the resolver fails

That's a good point. In that case, we can simplify the resolver factory and also skip the error here. Let me make some changes.

Reply:
Done